### PR TITLE
fixed bybit stop_loss triggerDirection for Shorts order

### PR DESCRIPTION
**Pull request by evanherve**
stop loss for short orders raised Bybit error where triggerDirection was not correct

### CODE REVIEW

**Comment by carlosmiei** [comment]
Hello @evanherve,
thanks for your contribution.
Can you revert all files except `bybit.ts`? (we don't need to commit auto-generated files)

**Comment by evanherve** [comment]
Hi @carlosmiei . Done

File: ts/src/bybit.ts [hunk 1]
```diff
@@ -3496,18 +3496,18 @@ export default class bybit extends Exchange {
...
         const isTakeProfit = takeProfit !== undefined;
+        const isBuy = side === 'buy';
+        const ascending = stopLossTriggerPrice ? !isBuy : isBuy;
         if (triggerPrice !== undefined) {
-            const isBuy = side === 'buy';
-            const ascending = stopLossTriggerPrice ? !isBuy : isBuy;
             request['triggerDirection'] = ascending ? 2 : 1;
             request['triggerPrice'] = this.priceToPrecision (symbol, triggerPrice);
         } else if (isStopLossTriggerOrder || isTakeProfitTriggerOrder) {
             if (isStopLossTriggerOrder) {
```
Comment (frosty00) [inline ts/src/bybit.ts:3505]: can we refactor this if clause

Comment (frosty00) [reply ts/src/bybit.ts:3505]: seems like it's a bit unnecessary, the code could be

```
request['triggerDirection'] = ascending ? 2 : 1;
const triggerPrice = isStopLossTriggerOrder ? stopLossTriggerPrice : takeProfitTriggerPrice
request['triggerPrice'] = this.priceToPrecision (symbol, triggerPrice);
```

Comment (Dan-krm) [reply ts/src/bybit.ts]: I changed this slightly but left the conditional so trigger orders can be used to open a position but stopLossTrigger and takeProfitTrigger orders are reduceOnly

**Pull request merged**